sources/core/Stride.Core.DiagnosticsServices/PropertyAttributeFinder.cs

```diff
@@ -0,0 +1,48 @@
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using System.Reflection;
+using Microsoft.CodeAnalysis.CSharp;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+
+namespace StrideDiagnostics;
+internal class PropertyAttributeFinder
+{
+    /// <summary>
+    /// Walks through a base class of a class and retrieves all allowed Properties.
+    /// Then it tries to get it's own base class and get from it all allowed Properties recursively.
+    /// All the Properties get summed up.
+    /// If they fulfill :
+    ///     Proper Accessors <see cref="PropertyHasAllowedAccessors(IPropertySymbol)"/>
+    ///     Shouldnt be Ignored <see cref="HasDataMemberIgnoreAttribute(IPropertySymbol)"/>
```

Comment:
Is `HasDataMemberIgnoreAttribute` method missing?

Reply:
Yes i moved it to the PropertyAttributeFinderExtensions, will remove it

Reply:
https://github.com/stride3d/stride/pull/1778/commits/7e5ef6e33c2ae8223e8e5e7258b4ce49b8ef389b

was removed as its unnecessary anyway